src/Nethermind/Nethermind.Evm/VirtualMachine.cs

```diff
@@ -2017,6 +2017,69 @@ void UpdateMemoryCost(in UInt256 position, in UInt256 length)
 
                         break;
                     }
+                    case Instruction.TLOAD:
+                    {
+                        if (!spec.TransientStorageEnabled)
+                        {
+                            EndInstructionTraceError(EvmExceptionType.BadInstruction);
+                            return CallResult.InvalidInstructionException;
+                        }
+                        Metrics.TloadOpcode++;
+                        var gasCost = spec.GetTLoadCost();
+
+                        if (!UpdateGas(gasCost, ref gasAvailable))
+                        {
+                            EndInstructionTraceError(EvmExceptionType.OutOfGas);
+                            return CallResult.OutOfGasException;
+                        }
+
+                        stack.PopUInt256(out UInt256 storageIndex);
+                        StorageCell storageCell = new(env.ExecutingAccount, storageIndex);
+
+                        byte[] value = _storage.GetTransientState(storageCell);
+                        stack.PushBytes(value);
+
```

Comment:
missing tracing:

```
                        if (_txTracer.IsTracingOpLevelStorage)
                        {
                            _txTracer.LoadOperationStorage(storageCell.Address, storageIndex, value);
                        }
```

Also we need to somehow tell tracer its transient

Reply:
It looks like it's Storage level tracing is only being used in the `GethLikeTxTracer`. One solution is to add a new `TransientStorage` property to `GethTxTraceEntry` such that it gets serialized with the rest of the output on `BuildResult()`. Without the implementation in Geth or specification in the EIP spec, this seems like a reasonable user experience. WDYT?

Reply:
Why not add something to the main interface and then let the implementer decide if he will handle it? Sometimes people write custom tracers in plugins to do custom things.